src/main/java/net/snowflake/ingest/streaming/internal/DataValidationUtil.java

```diff
@@ -957,6 +957,82 @@ static double validateAndParseReal(String columnName, Object input, long insertR
         columnName, input.getClass(), "REAL", new String[] {"Number", "String"}, insertRowIndex);
   }
 
+  /**
+   * Validates and parses input Iceberg INT column. Allowed Java types:
+   *
+   * <ul>
+   *   <li>Number
+   *   <li>String
+   * </ul>
+   *
+   * @param columnName Column name, used in validation error messages
+   * @param input Object to validate and parse
+   * @param insertRowIndex Row index for error reporting
+   * @return Parsed integer
+   */
+  static int validateAndParseIcebergInt(String columnName, Object input, long insertRowIndex) {
```

Comment:
Should we add unit tests for these functions, which focus on corner cases? Min and max values, Double/Float NaN+positive/negative infinity, big integers and big decimals outside of the allowed range, etc.?

Reply:
Added tests [here](https://github.com/snowflakedb/snowflake-ingest-java/blob/21afa2b86f4f4523f905fab3ac5ad622c4d62cd3/src/test/java/net/snowflake/ingest/streaming/internal/DataValidationUtilTest.java#L1200).